Python/Product/PythonTools/PythonTools/Environments/AddVirtualEnvironmentOperation.cs

```diff
@@ -0,0 +1,186 @@
+            _options = _site.GetComponentModel().GetService<IInterpreterOptionsService>();
+            _logger = _site.GetService(typeof(IPythonToolsLogger)) as IPythonToolsLogger;
+        }
+
+        public async Task RunAsync() {
+            var outputWindow = OutputWindowRedirector.GetGeneral(_site);
+            var taskHandler = _statusCenter.PreRegister(
+                new TaskHandlerOptions() {
+                    ActionsAfterCompletion = CompletionActions.RetainAndNotifyOnFaulted | CompletionActions.RetainAndNotifyOnRanToCompletion,
+                    Title = Strings.VirtualEnvStatusCenterCreateTitle.FormatUI(PathUtils.GetFileOrDirectoryName(_virtualEnvPath)),
+                    DisplayTaskDetails = (t) => { outputWindow.ShowAndActivate(); }
+                },
+                new TaskProgressData() {
+                    CanBeCanceled = false,
+                    ProgressText = Strings.VirtualEnvStatusCenterCreateProgressPreparing,
+                    PercentComplete = null,
+                }
+            );
+
+            var task = CreateVirtualEnvironmentAsync(taskHandler);
+            taskHandler.RegisterTask(task);
+            _site.ShowTaskStatusCenter();
+        }
+
+        private async Task CreateVirtualEnvironmentAsync(ITaskHandler taskHandler) {
+            IPythonInterpreterFactory factory = null;
+
+            bool failed = false;
+            var baseInterp = _registry.FindInterpreter(_baseInterpreter);
+
+            try {
+                factory = await VirtualEnv.CreateAndAddFactory(
+                    _site,
+                    _registry,
+                    _options,
+                    _project,
+                    _virtualEnvPath,
+                    baseInterp,
+                    _registerAsCustomEnv,
+                    _customEnvName,
+                    _useVEnv
+                );
+
+                if (factory != null) {
+                    if (_installReqs && File.Exists(_reqsPath)) {
+                        await InstallPackagesAsync(taskHandler, factory);
+                    }
+
+                    await _site.GetUIThread().InvokeTask(async () => {
+                        if (_project != null) {
+                            _project.AddInterpreter(factory.Configuration.Id);
+                            if (_setAsCurrent) {
+                                _project.SetInterpreterFactory(factory);
+                            }
+                        }
+
+                        if (_setAsDefault && _options != null) {
+                            _options.DefaultInterpreter = factory;
+                        }
+
+                        if (_viewInEnvWindow) {
+                            await InterpreterListToolWindow.OpenAtAsync(_site, factory);
+                        }
+                    });
+                }
+            } catch (Exception ex) when (!ex.IsCriticalException()) {
+                failed = true;
+                throw;
```

Comment:
You could make this `failed = false` at the end of the `try` block, and then remove the `catch` entirely.